### PR TITLE
Ta1 forever loop sganes21

### PR DESCRIPTION
**Pull request by sganes21**
Added classes and functions to organize dataframe
Added matplotlib to display scatterplot

### CODE REVIEW

File: api_pubmed.py [hunk 1]
```diff
@@ -58,8 +95,16 @@ def main():
     for year in range(int(args.start), int(args.end)):
         for month in range(1, 13):
             for disease in args.diseases.split(','):
-                pprint(pubmed.query(construct_term(year, month, disease)))
+                rawdata=pubmed.query(construct_term(year, month, disease))
+                data_aggregator.add_raw_data(rawdata, year, month,disease)
+                # pprint(pubmed.query(construct_term(year, month, disease)))
```
Comment (haranku16) [inline api_pubmed.py]: Can we clean this up?

File: api_pubmed.py [hunk 2]
```diff
@@ -50,6 +52,41 @@ def construct_term(year, month, disease):
...
+
+    def add_raw_data(self,rawdata,year,month,disease):
+
+        self.count.append(int(rawdata["esearchresult"]["count"]))  
+        self.year_month.append(str(year) + "/" + str(month))
+        self.disease.append(disease)
+
+    def build_data_dictionary(self):
+   
+        return {"year_month":self.year_month, "count": self.count, "disease":self.disease}
```
Comment (haranku16) [inline api_pubmed.py:86]: nitpick: can we been consistent with the spacing around the colons?

File: api_pubmed.py [hunk 3]
```diff
@@ -50,6 +52,41 @@ def construct_term(year, month, disease):
     '''
     return f'%28%28"{year}%2F{month}%2F01"%5BDate+-+Publication%5D+%3A+"{year if month < 12 else year + 1}%2F{month + 1 % 12}%2F01"%5BDate+-+Publication%5D%29%29+AND+%28{disease}%5BTitle%2FAbstract%5D%29'
 
+
+def create_dataframe(data):
+ 
+    '''
+    Creating dataframe of scraped pubmed data for visulaization
+    '''
+    df= pd.DataFrame(data)
```
Comment (haranku16) [inline api_pubmed.py]: We can directly return this. The variable isn't necessary.

File: api_pubmed.py [hunk 4]
```diff
@@ -50,6 +52,41 @@ def construct_term(year, month, disease):
...
+   
+        return {"year_month":self.year_month, "count": self.count, "disease":self.disease}
+
+
+
+
+data_aggregator = DataAggregator()
+
+
+
```
Comment (haranku16) [inline api_pubmed.py]: Let's clean up the extra newlines so that there's at most one space between lines of code.

File: api_pubmed.py [hunk 5]
```diff
@@ -50,6 +52,41 @@ def construct_term(year, month, disease):
...
+def create_dataframe(data):
+ 
+    '''
+    Creating dataframe of scraped pubmed data for visulaization
+    '''
+    df= pd.DataFrame(data)
+    return df
+
+
+class DataAggregator:
```
Comment (haranku16) [inline api_pubmed.py:63]: Can we add docstrings to this class and its methods?

File: api_pubmed.py (hunk 3 above)
Comment (haranku16) [inline api_pubmed.py]: np: let's add a space before the `=`

**Review by haranku16 — CHANGES_REQUESTED** [review]
Left a few stylistic comments. Otherwise looks good!

File: api_pubmed.py (hunk 1 above)
Comment (sganes21) [reply api_pubmed.py]: Removed commented statement

File: api_pubmed.py (hunk 2 above)
Comment (sganes21) [reply api_pubmed.py:86]: Reduced the spacing

File: api_pubmed.py (hunk 3 above)
Comment (sganes21) [reply api_pubmed.py]: Removed the variable

File: api_pubmed.py (hunk 4 above)
Comment (sganes21) [reply api_pubmed.py]: Removed the new lines

File: api_pubmed.py (hunk 5 above)
Comment (sganes21) [reply api_pubmed.py:63]: Added docstrings to class and methods

File: api_pubmed.py (hunk 3 above)
Comment (sganes21) [reply api_pubmed.py]: Removed unnecessary variable

**Review by haranku16 — APPROVED** [review]
Looks good! Thanks for getting the scatter plot working!!

**Pull request merged**